D1673/P1673.bs

```diff
@@ -5088,61 +5064,77 @@ satisfy the following requirements.
 All of these requirements presume that the algorithm's asymptotic complexity requirements,
 if any, are satisfied.
 
-1. Any input or output `mdspan`'s `value_type` meets the requirements of `semiregular`.
-
-2. The algorithm may perform read-only access on any input or output `mdspan`
-    arbitrarily many times.
-
-3. The algorithm or method may make arbitrary instances or copies
-    of any input or output `mdspan`'s `value_type`.
-
-4. The algorithm may assign arbitrarily many times to any reference
-    resulting from a valid output `mdspan` access,
-    as long as the algorithm's asymptotic complexity requirements are satisfied.
-
-5. The algorithm may assign any addend
-    in the algorithm's mathematical expression
-    to any reference resulting from a valid output `mdspan` access.
-
-6. The algorithm may reorder addends in its mathematical expression arbitrarily.
+1. Any input or output `mdspan`'s `value_type`
+    meets the requirements of `semiregular`.
+
+2. The algorithm may perform read-only access
+    on any input or output `mdspan` arbitrarily many times.
+
+3. The algorithm or method may make arbitrary many objects
+    of any input or output `mdspan`'s `value_type`,
+    value-initializing them or direct-initializing them
+    with any existing object of that type.
+
+4. The algorithm may assign arbitrarily many times
+    to any reference
+    resulting from a valid output `mdspan` access.
+
+5. If the algorithm's mathematical expression
+    uses addition,
+    but not subtraction, multiplication, or division,
+    then the algorithm evaluates the mathematical expression
+    using a sequence of evaluations of
+    `+`, `+=`, and `=` operators
+    that would produce the correct result
+    when operating on elements of a commutative semigroup.
```

Comment:
I think "exact arithmetic" _would_ work here: it doesn't require that implementations _use_ exact arithmetic (or test with it), because we know from algebra what transformations would preserve the value of the "textbook expression".  However, using the algebraic terminology might be superior in that it specifies the permissible operations other than implicitly via the allowed operators.

Reply:
> I think "exact arithmetic" would work here....

In this particular case of only `+`, `+=`, and `=`, I think it would.  I've been trying to avoid that phrase for anything involving `*`, because it might incorrectly suggest the commutative multiplication of real numbers.